Better handeling of submodules

dbeniamine opened the pull request:
Hello,

Your plugin is quite nice, I added a few things to manage properly git
submodules (clean commits and move file).

I also plan to add a remove function and create the vim documentation.

Hoping you'll be interested by my work,
Bests regards,

David

dbeniamine committed:
Add: proper submodule handling

dbeniamine committed:
Readme updated

dbeniamine committed:
Documentation

ADD: Vim help.txt documentation from Readme
CHG: Readme Updated
ADD: Gitignore

dbeniamine committed:
Destruct command to remove definitively a plugin

dbeniamine committed:
Fix badlinks in doc

dbeniamine committed:
fixed yet another bad link

dbeniamine commented:
Hum I already added the doc and the Destruct command (It was quicker than what I
thought), still I haven't tested Destruct without git submodule yet.

dbeniamine committed:
Upgrade command

It is now possible to upgrade vim plugins using :Upgrade command

dbeniamine committed:
Upgrade: small fix, need more tests

dbeniamine committed:
Let the user set the max Scry results displayed

ADD: Change number of results displayed by Scry search using
g:VizardryNbScryResults.
CHG: Clean the &lt;, &gt;, <b>, </b> in the vim documentation

dbeniamine committed:
Display Readme from invoke command

ADD: Display the README from Invoke by hitting 'd'
FIX: Bug on Invoke call introduced in previous commit

dbeniamine committed:
missing extractor script

dbeniamine commented:
Well it seems that I added a few others features: 

+ Upgrade command (actually I had the code since a while but I was too lazy to
create a full plugin), works also with vim.org plugins, the user just have to
add a two lines files (script url, version number (0 for init))
+ Possibility to look at the Readme in Invoke command
+ Possibility to set the Scry result list length

Hope you'll like it :)

dbeniamine committed:
Better readme retrieving

dbeniamine committed:
Clear screen before displaying readme

dbeniamine committed:
View Readmes in a new vim instance

dbeniamine committed:
Command name modified

To fit the naming policy of the plugin, Upgrade is now Evolve and Destruct is
now known as Vanish.

Also documentation updated

dbeniamine committed:
Updated vimorg script name

dbeniamine committed:
Echo plugin name instead of path in Evolve commits

dbeniamine committed:
fix bad link in readme

dbeniamine committed:
Vanish: force deinit

CHG: force submodule deinit during Vanish: without force, deinit would be
cancelled if tags have been created by helptags ...

ardagnir commented:
This looks really awesome!
There's a lot here and I'm really busy right now, but I'll try to look it over soon.

dbeniamine committed:
Navigate through results using Invoke

CHG: We can now navigate easily through the search results using :Invoke
FIX: :Helptags wasn't called at the right time

dbeniamine committed:
Documentation re organized

A little more formal documentation (but not that much)

dbeniamine committed:
Fix mistakes in doc

dbeniamine committed:
microscopic interface change + doc

ardagnir commented on plugin/vizardry.vim:
```diff
@@ -384,6 +502,9 @@ endfunction
 let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')
+let s:EvolveVimOrgPath = s:scriptDir.'/EvolveVimOrgPlugins.sh'
+let s:relativeBundleDir=substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
+let s:relativeBundleDir=substitute(s:relativeBundleDir,'^/','','')
```
These two previous lines break vizardry if you're not using submodules.

dbeniamine replied:
Those lines ?

```
let s:scriptDir = expand('<sfile>:p:h')
let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')
```

Aren't they exactly the same in your version ?
Can you tell me more about the bug so I can fix it ?

ardagnir replied:
No, lines 506 and 507

Error detected while processing /home/me/.vim/bundle/vizardry/plugin/vizardry.vim:
line  506:
E121: Undefined variable: g:VizardryGitBaseDir
E11Error detected while processing /home/me/.vim/bundle/vizardry/plugin/vizardry.vim:
line  506:
E121: Undefined variable: g:VizardryGitBaseDir
E116: Invalid arguments for function substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
E15: Invalid expression: substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
line  507:
E121: Undefined variable: s:relativeBundleDir
E116: Invalid arguments for function substitute(s:relativeBundleDir,'^/','','')
E15: Invalid expression: substitute(s:relativeBundleDir,'^/','','')
Press ENTER or type command to continue6: Invalid arguments for function substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
E15: Invalid expression: substitute(s:bundleDir,g:VizardryGitBaseDir,'','')
line  507:
E121: Undefined variable: s:relativeBundleDir
E116: Invalid arguments for function substitute(s:relativeBundleDir,'^/','','')
E15: Invalid expression: substitute(s:relativeBundleDir,'^/','','')
Press ENTER or type command to continue

dbeniamine replied:
Indeed the relativeBundleDir is useless if submodule are not used (Bundle dire relative to git base dir).
The bug is fixed thanks for the report.

dbeniamine commented:
Those lines ? 

    let s:scriptDir = expand('<sfile>:p:h')
    let s:bundleDir = substitute(s:scriptDir, '/[^/]*/[^/]*$', '', '')

Aren't they exactly the same in your version ?
Can you tell me more about the bug so I can fix it ?

dbeniamine committed:
Bug fix if not using submodules

Fix bug reported by ardagnir here:
https://github.com/ardagnir/vizardry/pull/5#discussion_r33417048

The relativeBundlePath variable was created and used even if submodule were
not used, crashing Vizardry. It should only exists and be used if submodule
method is chosen.

ardagnir commented:
I added a link to your fork in the readme

dbeniamine commented:
Thanks for the link :) .

dbeniamine committed:
Abort on connection error

CHG: If unable to retrieve plugin or version from vim.org, vizardry will now abort
without removing files.

dbeniamine committed:
Fix issue #4

FIX: Scry troubles managing empty descriptions in plugin list,
see [issue #4](https://github.com/ardagnir/vizardry/issues/4)

dbeniamine committed:
FIX: remove temporary files on failure

dbeniamine committed:
better error messages

dbeniamine committed:
Configure sort on github search results

dbeniamine committed:
Search plugins from specific users

ADD: possibility to search plugins by users with Invoke & Scry
FIX: Bug on Invoke introduced by commit 651f65b2d8cc0b79ae15b49320915ce77770f4f5

dbeniamine committed:
Prefer real bundle name to input nice for cloning

CHG: During Invoke propose to name the repo by its actual name at github
instead of the 'inputNice' which might not be so nice actually ...
CHG: Doc updated

dbeniamine committed:
Fix bug: search without username + small refactor

FIX: The detection of the '-u' parameter on Invoke and Scry was bad resulting
on failing search
RM: debug stuff slowing down vizardry
CHG: coding style avoid to use more than 80 cols

dbeniamine committed:
small doc update

dbeniamine committed:
More flexible search option

dbeniamine committed:
FIX: markdown error in readme

dbeniamine committed:
Refactor + Comment + Cleaner I/O

CHG: Dedicated echo function with standard colors.
CHG: Code commented and reorganized.
FIX: Several minor bug fix (including false error reports and poor handling of
wrong commands).

dbeniamine committed:
Fix indent issue on README

dbeniamine committed:
FIX small errors on Unbanish and Reload

dbeniamine committed:
Use autoload to speedup vim start

Moving as much things as possible to autoload to speedup vim startup.

The autoload directory is divided as follow:

autoload/
    vizardry.vim        Functions and variables used by most Vizardry commands
    vizardry/
        remote.vim      Functions/variables related to remote (aka github) work
        local.vim       Functions/variables which only work on local bundle
                        and that are not (are almost never) used by functions
                        from remote.vim

dbeniamine committed:
FIX Issue #1

FIX: Invoke error when using git clone
FIX: Banish bug: "function ListBanished not found"

dbeniamine committed:
Fix error when Invoking result 0 from Scry

dbeniamine committed:
Show Readme on Evolve

ADD: it is now possible to view a plugin's README after upgrading it with
:Evolve. Todo do so, add the following to your vimrc:
    let g:VizardryViewReadmeOnEvolve=1
`:Evolve` will then prompt you to show readme after each plugin upgrade

dbeniamine committed:
Call Invoke directly from Scry

ADD: After a call to Scry, Vizardry now prompt you to Invoke a plugin
CHG: Dialog method updated, see release notes for side effects

dbeniamine committed:
Small interface fix

FIX: Error message on bad Invoke script number
CHG: Same List number for Scry and Invoke

dbeniamine committed:
Show git log on evolve

dbeniamine committed:
Show only script name on evolve prompt

dbeniamine committed:
Cleaner prompt

dbeniamine committed:
Fix Evolve completion

dbeniamine committed:
Small interface improvements

FIX: Show a warning message when no results are found for a query
CHG: Show deprecated message when trying to evolve a path as a vim.org script

dbeniamine committed:
Huge documentation cleaning

dbeniamine committed:
Fix missing space in Readme

dbeniamine committed:
Readme table of contents + small fixes
